pg_search/src/query/mod.rs

```diff
@@ -317,6 +328,54 @@ impl SearchQueryInput {
                     )))
                 }
             }
+            Self::FuzzyPhrase {
+                field,
+                value,
+                distance,
+                transposition_cost_one,
+                prefix,
+                conjunction_mode,
+            } => {
+                let distance = distance.unwrap_or(2);
+                let transposition_cost_one = transposition_cost_one.unwrap_or(true);
+                let conjunction_mode = conjunction_mode.unwrap_or(true);
+                let prefix = prefix.unwrap_or(false);
+
+                let index_oid = config.index_oid;
+                let directory = WriterDirectory::from_index_oid(index_oid);
+                let search_index = SearchIndex::from_disk(&directory)
+                    .unwrap_or_else(|err| panic!("error loading index from directory: {err}"));
```

Comment:
I think we should try and avoid initializing a whole `SearchIndex` here. I've tried to design the query builders to be as decoupled as possible from the rest of the codebase, and the `SearchIndex` is a pretty hairy structure.

You might also be re-reading the search index from disk here, which shouldn't be necessary. It's initialized before this function is called, and I believe that's how the `Reader` parameter is passed in.

Both of the above are the intent behind the `field_lookup` and other parameters that you see passed in at the top of the function. Just trying to keep these query builders from "knowing" any more than necessary about other parts of our code.

I'm happy to help you think of a similar injection strategy for the tokenizer/token_stream.